src/interpreter/mod.rs

```diff
@@ -1261,10 +1432,24 @@ mod tests {
         interpreter.set_prog(prog);
 
         let value = interpreter.interpret().unwrap();
-        assert_eq!(value, Value::Boolean(true));
+        assert_eq!(
+            value,
+            ValueSet {
+                last_value: Value::Boolean(true),
+                used_values: vec![Value::Boolean(true)],
+                unused_values: vec![Value::Boolean(true)],
+            }
+        );
 
         let value = interpreter.interpret().unwrap();
-        assert_eq!(value, Value::Boolean(true));
+        assert_eq!(
+            value,
+            ValueSet {
+                last_value: Value::Boolean(true),
+                used_values: vec![Value::Boolean(true)],
+                unused_values: vec![Value::Boolean(true)],
+            }
+        );
```

Comment:
I think it'd be cleaner if all these tests cared only about `value.last_value` as before, so they don't test multiple things. Cases that test explicitly for used and unused values could be added. This way it'd be also easier to maintain them in case the structure changes.

Reply:
Makes sense to add tests specifically for the new structure and keep old tests on `last_value`, I'll do it.